docs-src/httproute.md

```diff
@@ -0,0 +1,303 @@
+
+### Rules
+
+Rules define semantics for matching an HTTP request based on conditions, optionally executing additional processing
+steps, and optionally forwarding the request to an API object.
+
+#### Matches
+
+Matches define conditions used for matching an HTTP request. Each match is independent, i.e. this rule will be matched if
+any single match is satisfied.
+
+Take the following matches configuration as an example:
+```yaml
+kind: HTTPRoute
+apiVersion: networking.x-k8s.io/v1alpha1
+...
+matches:
+  - path:
+      value: "/foo"
+    headers:
+      values:
+        version: "2"
+  - path:
+      value: "/v2/foo"
+```
+
+For a request to match against this rule, it must satisfy EITHER of the following conditions:
+
+ - A path prefixed with /foo **AND** contains the header "version: 2"
+ - A path prefix of /v2/foo
+
+If no matches are specified, the default is a prefix path match on “/”, which has the effect of matching every HTTP
+request.
+
+#### Filters (optional)
+
+Filters define processing steps that must be completed during the request or response lifecycle. Filters act as an
+extension point to express additional processing that may be performed in Gateway implementations. Some examples include
+request or response modification, implementing authentication strategies, rate-limiting, and traffic shaping.
+
+The following example adds header "my-header: foo" to HTTP requests with Host header "my.filter.com".
+```yaml
+kind: HTTPRoute
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: http-filter-1
+spec:
+  hostnames:
+    - my.filter.com
+  rules:
+    ...
+      filters:
+      - type: RequestHeaderModifier
+        requestHeaderModifier:
+          add:
+            my-header: foo
+```
+API conformance is defined based on the filter type. The effects of ordering multiple behaviors is currently
+unspecified. This may change in the future based on feedback during the alpha stage.
+
+Conformance levels are defined by the filter type:
+
+ - All "core" filters MUST be supported by implementations.
+ - Implementers are encouraged to support "extended" filters.
+ - "Custom" filters have no API guarantees across implementations.
+
+Specifying a core filter multiple times has unspecified or custom conformance.
+	
+#### ForwardTo (optional)
+
+ForwardTo defines API objects where matching requests should be sent. If unspecified, the rule performs no
+forwarding. If unspecified and no filters are specified that would result in a response being sent, a 503 error code is
+returned.
+
+The following example forwards all HTTP requests with Host header "my.filter.com" to service "my-filter-svc1":
+```yaml
+kind: HTTPRoute
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: http-filter-1
+spec:
+  hostnames:
+    - my.filter.com
+  rules:
+    - matches:
+      - path:
+          type: Prefix
+          value: /
+      forwardTo:
+      - serviceName: my-filter-svc1
+        port: 80
+```
```

Comment:
I think any complete example like this should probably come from the examples dir so we can ensure it doesn't become stale. Alternatively you could choose to highlight just the small part of this config relevant to `forwardTo`.